extension/src/helpers/muxedAddress.ts

```diff
@@ -0,0 +1,205 @@
+/**
+ * Utility functions for handling muxed addresses and memo logic
+ */
+import { NetworkDetails } from "@shared/constants/stellar";
+import { getContractSpec } from "@shared/api/internal";
+import { isContractId } from "popup/helpers/soroban";
+import {
+  isMuxedAccount,
+  isValidStellarAddress,
+  createMuxedAccount,
+  getBaseAccount,
+} from "helpers/stellar";
+
+export interface CheckContractMuxedSupportParams {
+  contractId: string;
+  networkDetails: NetworkDetails;
+}
+
+/**
+ * Checks if a contract's transfer function supports muxed addresses (CAP-0067)
+ * by examining the contract specification.
+ *
+ * A contract supports muxed addresses if the transfer function has a `to_muxed` parameter.
+ * This is the CAP-0067 pattern where the transfer signature is:
+ * transfer(from: Address, to_muxed: MuxedAddress, amount: i128)
+ *
+ * @param contractId - The contract ID to check
+ * @param networkDetails - Network details
+ * @returns Promise resolving to true if contract supports muxed addresses. false otherwise
+ */
+export async function checkContractMuxedSupport(
```

Comment:
NIT: since this is returning a boolean, would you mind using  the `is_` naming convention - something like `checkIsMuxedSupported`?